js/ui/marker.js

```diff
@@ -85,6 +89,98 @@ Marker.prototype = {
         return this._el;
     },
 
+    /**
+     * Binds a Popup to the Marker
+     * @param {HTMLElement|String|Popup} content the DOM content to appear in the popup, or 
+     *  an instance of the Popup class
+     * @param {Object} [options] options for the Popup class
+     * @param {boolean} [options.closeButton=true] If `true`, a close button will appear in the
+     *   top right corner of the popup.
+     * @param {boolean} [options.closeOnClick=true] If `true`, the popup will closed when the
+     *   map is clicked.
+     * @param {string} options.anchor - A string indicating the popup's location relative to
+     *   the coordinate set via [Popup#setLngLat](#Popup#setLngLat).
+     *   Options are `'top'`, `'bottom'`, `'left'`, `'right'`, `'top-left'`,
+     * `'top-right'`, `'bottom-left'`, and `'bottom-right'`.
+     * @returns {Marker} `this`
+     */
+
+    bindPopup: function(content, options){
+        if (content instanceof Popup) {
+            this._popup = content;
+        } else {
+            if (!this._popup || options) {
+                this._popup = new Popup(options);
+            }
+            content instanceof HTMLElement ? this._popup.setDOMContent(content) : this._popup.setHTML(content);
+        }
+            
+        if (this._popup && this._lngLat) this._popup.setLngLat(this._lngLat);
+
+        if (!this._popupHandlersAdded) {
+            this.getElement().addEventListener('click', this._openPopup.bind(this));
+            this._popupHandlersAdded = true;
+        }
+        return this;
+    },
+
+    /**
+     * Opens or closes the bound popup, depending on the current state
+     * @returns {Marker} `this`
+     */
+    togglePopup: function(){
+        if (this._popup) {
+            if (this._popup._map){
+                this._closePopup();
+            } else {
+                this._openPopup();
+            }
+        }
+    },
+
+    /**
+     * Returns the Popup instance that is bound to the Marker
+     * @returns {Popup} popup
+     */
+    getPopup: function(){
+        if (this._popup) {
+            return this._popup;
+        }
+    },
+
+    /**
+     * Unbinds a Popup from the Marker instance
+     * Returns the Popup instance that was bound to the Marker
+     * @returns {Popup} popup
+     */
+    unbindPopup: function(){
```

Comment:
The accessor convention we've been following in GL JS is to provide reset/unbind semantics via setters that accept `null`/`undefined`. If we applied that pattern here, it would look like:

* `marker.setPopup(popup)` to bind a popup
* `marker.getPopup()` (getter as is)
* `marker.setPopup(null)` to unbind a popup

I think I prefer the consistency with other parts of GL JS to consistency with Leaflet. WDYT?